src/mesh/RadioInterface.cpp

```diff
@@ -454,13 +454,13 @@ void RadioInterface::applyModemConfig()
 
     // If user has manually specified a channel num, then use that, otherwise generate one by hashing the name
     const char *channelName = channels.getName(channels.getPrimaryIndex());
-    int channel_num = (loraConfig.channel_num ? loraConfig.channel_num - 1 : hash(channelName)) % numChannels;
+    int channel_num = (loraConfig.channel_num ? loraConfig.channel_num : hash(channelName)) % numChannels;
 
     // Old frequency selection formula
     // float freq = myRegion->freqStart + ((((myRegion->freqEnd - myRegion->freqStart) / numChannels) / 2) * channel_num);
 
     // New frequency selection formula
-    float freq = myRegion->freqStart + (bw / 2000) + (channel_num * (bw / 1000));
+    float freq = myRegion->freqStart + (bw / 2000) + ((channel_num - 1) * (bw / 1000));
```

Comment:
This will shift the actual frequency even if we use the hash formula. Old behaviour: 1 is the lowest number, cause 0 in the protobuf means 'calculate it yourself'. Now: lowest number is 0. So freq will be different after the patch.

Reply:
ah, good you mention. that modulus operation threw me off. `channel_num` variable actually returns between 0 and (numChannels - 1).